### PR TITLE
Customized Logout

### PR DESCRIPTION
**Pull request by cetin-altiscale**
This is the initial implementation to customize logout behavior on Apache Ranger. When Altiscale Portal is used for user authentication, the logout action redirects to Altiscale Portal Logout page to clear sessions. Note that the logout behavior has a minor issue which is going to be fixed later.

### CODE REVIEW

File: security-admin/src/main/java/org/apache/ranger/rest/ServiceREST.java [hunk 1]
```diff
@@ -2892,6 +2892,22 @@ private RangerPolicyEngine getPolicyEngine(String serviceName) throws Exception
 	public String checkSSO() {
 		return String.valueOf(bizUtil.isSSOEnabled());
 	}
+
+	@GET
+	@Path("/checkssotype")
+	@Produces({ "application/json"})
```
Comment (zoran-altiscale) [inline security-admin/src/main/java/org/apache/ranger/rest/ServiceREST.java]: :) space. I thought I already trained you to see spaces :)

File: security-admin/src/main/resources/conf.dist/ranger-admin-site.xml [hunk 2]
```diff
@@ -258,6 +258,10 @@
 		<name>ranger.web.authentication.alt-kerberos.portal</name>
 		<value>https://localhost/sso/authorize</value>
 	</property>
+	<property>
+		<name>ranger.web.authentication.alt-kerberos.portal.logout</name>
+		<value>https://localhost/sso/logout</value>
```
Comment (zoran-altiscale) [inline security-admin/src/main/resources/conf.dist/ranger-admin-site.xml]: this is an example? Port number is missing, and I think ranger UI port isn't 80? I would even keep this empty and LOG.fatal if it's not set in the code...

File: security-admin/src/main/webapp/scripts/views/common/ProfileBar.js [hunk 3]
```diff
@@ -49,7 +49,7 @@ define(function(require){
 		events: function() {
 			var events = {};
 			//events['change ' + this.ui.input]  = 'onInputChange';
-			events['click ' + this.ui.logout]  = 'checkKnoxSSO';
+			events['click ' + this.ui.logout]  = 'checkSSO';
```
Comment (zoran-altiscale) [inline security-admin/src/main/webapp/scripts/views/common/ProfileBar.js:52]: this is altiscale-only change right?

File: security-admin/src/main/webapp/scripts/views/common/ProfileBar.js [hunk 4]
```diff
@@ -77,6 +77,32 @@ define(function(require){
...
+			that = this;
+			$.ajax({
+				url : url,
+				type : 'GET',
+				headers : {
+					"cache-control" : "no-cache"
+				},
+				success : function() {
+					if (checksso.altiSSO.isEnabled == 'true') {
+						window.location.replace(checksso.altiSSO.logoutURL);
```
Comment (zoran-altiscale) [inline security-admin/src/main/webapp/scripts/views/common/ProfileBar.js:91]: I still don't see where you invalidate session cookie?

File: security-admin/src/main/resources/conf.dist/ranger-admin-site.xml (hunk 2 above)
Comment (cetin-altiscale) [reply security-admin/src/main/resources/conf.dist/ranger-admin-site.xml]: Yes, this is a dummy example. We actually use our own portal's logout link in the deployment. I will fix and keep it empty if makes this clearer.

File: security-admin/src/main/webapp/scripts/views/common/ProfileBar.js (hunk 3 above)
Comment (cetin-altiscale) [reply security-admin/src/main/webapp/scripts/views/common/ProfileBar.js:52]: yes, this is only for altiscale.

File: security-admin/src/main/java/org/apache/ranger/rest/ServiceREST.java (hunk 1 above)
Comment (cetin-altiscale) [reply security-admin/src/main/java/org/apache/ranger/rest/ServiceREST.java]: :) you are right. It looks like I didn't pay attention this after copy-paste :( will fix this.

File: security-admin/src/main/webapp/scripts/views/common/ProfileBar.js (hunk 4 above)
Comment (cetin-altiscale) [reply security-admin/src/main/webapp/scripts/views/common/ProfileBar.js:91]: forgot to answer that. ranger uses the spring security framework and the session cookies are deleted on logout by the framework. the configuration is provided at https://github.com/apache/ranger/blob/c4020f89b929f323d133fbe81d2571deca29f934/security-admin/src/main/resources/conf.dist/security-applicationContext.xml#L58.

**Pull request merged**